### ISSUE TITLE
failing calls should return revert data

### ISSUE THREAD

**Issue by rainbreak** [dapphub/ds-proxy#19]
i.e. use `revert(0, response)` rather than `revert(0, 0)`. Needs a test adding as well.

**Comment by gbalabasquer** [comment]
I think it should be `revert(response, size)`. Regarding the test, is there a way to check the return value of a failing call with `dapp test`?

**Comment by rainbreak** [comment]
Yeah I think you'll have to construct something with assembly. Also possible this isn't in hevm yet :/

**Comment by gbalabasquer** [comment]
Change done, test as well.
However it is not returning any value with `returndatacopy` when the tx fails. It is possible is a `hevm` limitation as you mentioned.

**Comment by rainbreak** [comment]
Let's resolve this `hevm` issue before merging #18, just to be safe. I can take a look this week.
